tests/test_auth.py

```diff
@@ -0,0 +1,196 @@
+
+
+@mock.patch(f'{MODULE}.time', autospec=True)
+def test_jwt_payload_from_user_profile(mock_time, jwt_manager):
+    profile = UserProfile(
+        user_id='test_user_id',
+        token='test_token',
+        groups=['test_group1', 'test_group2'],
+        first_name='test_first_name',
+        last_name='test_last_name',
+        email='test_email',
+    )
+    mock_time.return_value = 1
+
+    jwt_manager.session_ttl = 5
+    payload = jwt_manager._jwt_payload_from_user_profile(profile)
+
+    assert payload == {
+        'urs-user-id': 'test_user_id',
+        'first_name': 'test_first_name',
+        'last_name': 'test_last_name',
+        'email': 'test_email',
+        'urs-access-token': 'test_token',
+        'urs-groups': ['test_group1', 'test_group2'],
+        'iat': 1,
+        'exp': 6
+    }
+
+
+@mock.patch(f'{MODULE}.JwtManager._in_blacklist')
+def test_get_profile_from_header(mock_in_blacklist, jwt_manager, jwt_priv_key):
+    mock_in_blacklist.return_value = False
+    payload = {
+        'urs-user-id': 'test_user',
+        'first_name': 'test',
+        'last_name': 'user',
+        'email': 'user@emailwebsite.com',
+        'urs-access-token': 'foo',
+        'urs-groups': []}
+
+    headers = {
+        'Cookie': f'auth-cookie={jwt.encode(payload, jwt_priv_key, "RS256")}'
+    }
+    user_profile = jwt_manager.get_profile_from_headers(headers)
+    assert user_profile.user_id == 'test_user'
+
+    headers = {}
+    assert jwt_manager.get_profile_from_headers(headers) is None
+
+
+@mock.patch(f'{MODULE}.JwtManager._in_blacklist')
+def test_get_profile_from_header_jwt_blacklisted(mock_in_blacklist, jwt_manager, jwt_priv_key):
+    mock_in_blacklist.return_value = True
+    payload = {
+        'urs-user-id': 'test_user',
+        'first_name': 'test',
+        'last_name': 'user',
+        'email': 'user@emailwebsite.com',
+        'urs-access-token': 'foo',
+        'urs-groups': []}
+
+    headers = {
+        'Cookie': f'auth-cookie={jwt.encode(payload, jwt_priv_key, "RS256")}'
+    }
+    user_profile = jwt_manager.get_profile_from_headers(headers)
+    assert user_profile is None
+
+
+@mock.patch(f'{MODULE}.JwtManager._encode_jwt')
+@mock.patch(f'{MODULE}.time')
+def test_get_header_to_set_auth_cookie(
+        mock_time,
+        mock_encode_jwt,
+        jwt_manager,
+        jwt_priv_key,
+):
+    jwt_manager.cookie_name = 'auth-cookie'
+    jwt_manager.session_ttl = 1
+
+    mock_encode_jwt.return_value = 'COOKIE_VALUE'
+    mock_time.return_value = 0
+
+    profile = UserProfile(
+        user_id='test_user_id',
+        token='test_token',
+        groups=['test_group1', 'test_group2'],
+        first_name='test_first_name',
+        last_name='test_last_name',
+        email='test_email',
+        iat=0,
+        exp=0
+    )
+
+    header = jwt_manager.get_header_to_set_auth_cookie(profile, '')
+    assert header == {'SET-COOKIE': 'auth-cookie=COOKIE_VALUE; Expires=Thu, 01 Jan 1970 00:00:01 GMT; Path=/'}
```

Comment:
How is the expiration time 1 second past midnight? Shouldn't it be right at 0?

Reply:
line 170 session_ttl is 1